packages/iris-grid/src/IrisGridModelUpdater.tsx

```diff
@@ -40,164 +41,163 @@ interface IrisGridModelUpdaterProps {
-    totalsConfig = null,
-    selectDistinctColumns = EMPTY_ARRAY,
-    pendingRowCount = 0,
-    pendingDataMap = EMPTY_MAP,
-    frozenColumns,
-    formatColumns,
-    columnHeaderGroups,
-    partitionConfig,
-  }: IrisGridModelUpdaterProps) => {
-    const columns = useMemo(
-      () =>
-        IrisGridUtils.getModelViewportColumns(
-          modelColumns,
-          left,
-          right,
-          movedColumns,
-          hiddenColumns,
-          alwaysFetchColumns,
-          COLUMN_BUFFER_PAGES
-        ),
-      [
-        modelColumns,
+function IrisGridModelUpdater({
+  model,
+  top,
+  bottom,
+  left,
+  right,
+  filter,
+  formatter,
+  reverseType = TableUtils.REVERSE_TYPE.NONE,
+  sorts,
+  customColumns,
+  movedColumns,
+  hiddenColumns,
+  alwaysFetchColumns,
+  rollupConfig = null,
+  totalsConfig = null,
+  selectDistinctColumns = EMPTY_ARRAY,
+  pendingRowCount = 0,
+  pendingDataMap = EMPTY_MAP,
+  frozenColumns,
+  formatColumns,
+  columnHeaderGroups,
+  partitionConfig,
+  showExtraGroupColumn,
+}: IrisGridModelUpdaterProps): JSX.Element | null {
+  // Check for showExtraGroupColumn before memoizing columns, since updating it will change the columns
+  useOnChange(() => {
+    if (isIrisGridTreeTableModel(model) && showExtraGroupColumn != null) {
+      model.showExtraGroupColumn = showExtraGroupColumn;
+    }
+  }, [model, showExtraGroupColumn]);
+
+  const columns = useMemo(
+    () =>
+      IrisGridUtils.getModelViewportColumns(
+        model.columns,
         left,
         right,
         movedColumns,
         hiddenColumns,
         alwaysFetchColumns,
-      ]
-    );
-
-    useEffect(
-      function updateFilter() {
-        model.filter = filter;
-      },
-      [model, filter]
-    );
-    useEffect(
-      function updateSorts() {
-        const sortsForModel = [...sorts];
-        if (reverseType !== TableUtils.REVERSE_TYPE.NONE) {
-          sortsForModel.push(model.dh.Table.reverse());
-        }
-        model.sort = sortsForModel;
-      },
-      [model, sorts, reverseType]
-    );
-    useEffect(
-      function updateFormatter() {
-        model.formatter = formatter;
-      },
-      [model, formatter]
-    );
-    useEffect(
-      function updateCustomColumns() {
-        if (model.isCustomColumnsAvailable) {
-          model.customColumns = customColumns;
-        }
-      },
-      [model, customColumns]
-    );
-    useEffect(
-      function updateFormatColumns() {
-        if (model.isFormatColumnsAvailable) {
-          model.formatColumns = formatColumns;
-        }
-      },
-      [model, formatColumns]
-    );
-    useEffect(
-      function updateViewport() {
-        model.setViewport(top, bottom, columns);
-      },
-      [model, top, bottom, columns]
-    );
-    useEffect(
-      function updateRollupCOnfig() {
-        if (model.isRollupAvailable) {
-          model.rollupConfig = rollupConfig;
-        }
-      },
-      [model, model.isRollupAvailable, rollupConfig]
```

Comment:
I'm confused why the eslint rule would want this for `useEffect`, but not for `useOnChange` as a custom exhaustive deps hook. Should add it back so this doesn't accidentally break in some way